test.js

```diff
@@ -569,3 +569,16 @@ if (Promise.prototype.finally) {
 		t.is(result.message, 'called');
 	});
 }
+
+test('the spawned process is cancelable', t => {
+	const spawned = execa('ls');
+	spawned.catch(error => {
+		if (error.message === 'reason' && error.isCanceled) {
+			return;
+		}
+
+		throw error;
+	}); // To handle the CancelError
+	spawned.cancel('reason');
```

Comment:
I think this test should be split into several tests each testing different parts of the PR:
  - test that after `cancel()` `result.canceled` is `true`
  - test that without `cancel()` `result.canceled` is `false`
  - test that after `cancel()` `promise is rejected with the right error
  - test that the message passed as a `reason` appears somewhere inside `result.message`

Reply:
I'd have to explore ava's docs, I never written tests before :sweat_smile:

Reply:
If you've never written tests before, I think this might a great opportunity for you here. [Ava](https://github.com/avajs/ava) is a great project and it's beginner-friendly.

For me the main thing is:
  - keeping tests simple: each test does one thing, then check if the expected result is correct.
  - splitting tests into small tests that each test one thing.
  - keeping tests independent from each other. Ava runs tests in parallel by default so it enforces avoiding using any global state.
  - using the tools the test runner provides (here Ava), e.g. `t.throwsAsync()` assertions, etc.
  - making sure you're covering all the features you're adding and things that could break.
  - remembering target platform: Windows/Mac/Linux, Node.js `>= 8`

Please ask any questions you might have while learning Ava and writing those tests, we're here to help :)

Reply:
thanks a lot! actually I am a bit busy right now. I'll do all this tonight (5-6 hours from now)

Reply:
There is no rush :+1: